File: poll-app-server/src/main/java/poll/app/services/NotificationService.java
package poll.app.services;

import java.util.List;

import poll.app.models.Notification;
import poll.app.models.Poll;

public interface NotificationService
{
        List<Notification> findByUser ();

        boolean create (Poll poll);

        boolean update ();

        List<Notification> readByUser();
}

File: poll-app-client/src/app/notification.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Notification } from '../app/models/notification.model';
import { environment } from '../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class NotificationService {

  setNotificationsRead() {
    return this.http.patch<Notification[]>(environment.apiUrl + '/api/notification/', null);
  }

  constructor(private http: HttpClient) { }

  public findByUser() {
    return this.http.get<Notification[]>(environment.apiUrl + '/api/notification')
  }
}
